lib/IRGen/GenDistributed.cpp

```diff
@@ -588,9 +589,26 @@ static llvm::Value *lookupWitnessTable(IRGenFunction &IGF, llvm::Value *witness,
   assert(Lowering::TypeConverter::protocolRequiresWitnessTable(protocol));
 
   auto &IGM = IGF.IGM;
-  auto *protocolDescriptor = IGM.getAddrOfProtocolDescriptor(protocol);
+  llvm::Value *protocolDescriptor = IGM.getAddrOfProtocolDescriptor(protocol);
+
+  bool signedProtocolDescriptor = IGM.getAvailabilityRange().isContainedIn(
+    IGM.Context.getSignedConformsToProtocolAvailability());
+
+  auto conformsToProtocolFunctionPointer = signedProtocolDescriptor ?
+    IGM.getConformsToProtocol2FunctionPointer() :
+    IGM.getConformsToProtocolFunctionPointer();
+
+  // Sign the protocol descriptor.
+  auto schema = IGF.IGM.getOptions().PointerAuth.ProtocolDescriptorsAsArguments;
+  if (schema && protocolDescriptor) {
```

Comment:
s/protocolDescriptor/signedProtocolDescriptor

We want to sign if we call the new stdlib function.